src/cpp/core/system/PosixSystem.cpp

```diff
@@ -1264,19 +1298,49 @@ core::Error pidof(const std::string& process, std::vector<PidType>* pPids)
    return Success();
 }
 
-Error processInfo(const std::string& process, std::vector<ProcessInfo>* pInfo)
+Error processInfo(const std::string& process, std::vector<ProcessInfo>* pInfo, ProcessFilter filter)
 {
-   std::vector<PidType> pids;
-   Error error = pidof(process, &pids);
+   // use ps to capture process info
+   // output format
+   // USER:PID:PPID:PGID:PROCNAME
+   // we use a colon as the separator as it is not a valid path character in OSX
+   std::string cmd = process.empty() ? "ps acxj | awk '{OFS=\":\"; print $1,$2,$3,$4,$10}'"
```

Comment:
I think this would be less fragile if implemented as a pure ps format string rather than piping to awk (maybe there's a good reason this is not the case, let me know what you think)